File: Python/refer-offer/er_cha_shu_de_shen_du_lcof.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def maxDepth(self, root: TreeNode) -> int:
        self.res = 0
        def dfs(root: TreeNode, depth: int):
            if not root: return
            dfs(root.left, depth + 1)
            self.res = max(self.res, depth)
            dfs(root.right, depth + 1)
        dfs(root, 1)
        return self.res

File: core/tmp/Java/src/com/dbc/algorithm/ReferOffer/ErChaShuDeShenDuLcof.java
package com.dbc.algorithm.ReferOffer;

public class ErChaShuDeShenDuLcof {
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode(int x) {
            val = x;
        }
    }

    private int res = 0;
    public int maxDepth(TreeNode root) {
        dfs(root, 1);
        return res;
    }
    public void dfs(TreeNode root, int depth) {
        if (root == null) return;
        dfs(root.left, depth + 1);
        res = Math.max(res, depth);
        dfs(root.right, depth + 1);
    }
}
